packages/common-utils/src/renderChartConfig.ts

```diff
@@ -688,18 +727,119 @@ function renderLimit(
   return chSql`${{ Int32: chartConfig.limit.limit }}${offset}`;
 }
 
-export async function renderChartConfig(
+// CTE (Common Table Expressions) isn't exported at this time. It's only used internally
+// for metric SQL generation.
+type ChartConfigWithOptDateRateAndCte = ChartConfigWithOptDateRange & {
+  with?: { name: string; sql: ChSql }[];
+};
+
+function renderWith(
+  chartConfig: ChartConfigWithOptDateRateAndCte,
+  metadata: Metadata,
+): ChSql | undefined {
+  const { with: withClauses } = chartConfig;
+  if (withClauses) {
+    return concatChSql(
+      '',
+      withClauses.map(clause => chSql`WITH ${clause.name} AS (${clause.sql})`),
+    );
+  }
+
+  return undefined;
+}
+
+function translateMetricChartConfig(
   chartConfig: ChartConfigWithOptDateRange,
+): ChartConfigWithOptDateRateAndCte {
+  const metricTables = chartConfig.metricTables;
+  if (!metricTables) {
+    return chartConfig;
+  }
+
+  // assumes all the selects are from a single metric type, for now
+  const { select, from, ...restChartConfig } = chartConfig;
+  if (!select || !Array.isArray(select)) {
+    throw new Error('multi select or string select on metrics not supported');
+  }
+
+  const { metricType, metricName, ..._select } = select[0]; // Initial impl only supports one metric select per chart config
+  if (metricType === MetricsDataType.Gauge && metricName) {
+    return {
+      ...restChartConfig,
+      select: [
+        {
+          ..._select,
+          valueExpression: 'Value',
+        },
+      ],
+      from: {
+        ...from,
+        tableName: metricTables[MetricsDataType.Gauge],
+      },
+      where: `MetricName = '${metricName}'`,
+      whereLanguage: 'sql',
+    };
+  } else if (metricType === MetricsDataType.Sum && metricName) {
+    return {
+      ...restChartConfig,
+      with: [
+        {
+          name: 'RawSum',
+          sql: chSql`SELECT MetricName,Value,TimeUnix,Attributes,
```

Comment:
We need to expose all the original columns here, otherwise the user won't be able to query the table for something like `GROUP BY ResourceAttrributes['host.name']`